presto-spi/src/main/java/com/facebook/presto/spi/plan/FilterStatsCalculatorService.java

```diff
@@ -27,5 +27,5 @@
  */
 public interface FilterStatsCalculatorService
 {
-    TableStatistics filterStats(TableStatistics tableStatistics, RowExpression predicate, ConnectorSession session, Map<ColumnHandle, String> columnNames, Map<ColumnHandle, Type> columnTypes);
+    TableStatistics filterStats(TableStatistics tableStatistics, RowExpression predicate, ConnectorSession session, Map<ColumnHandle, String> columnNames, Map<String, Type> columnTypes);
```

Comment:
Two questions:
- Could you add a javadoc to explain the reason to use strings? (/* because column handle contains all other unnecessary fields blah blah blah */)
- Maybe removing `requiredSubfields` from the equal function of hive column handle can solve this issue? But it does seem a bit hacky.

Reply:
@highker James, thank you for review.

> Could you add a javadoc to explain the reason to use strings? (/* because column handle contains all other unnecessary fields blah blah blah */)

Will do.

> Maybe removing requiredSubfields from the equal function of hive column handle can solve this issue? But it does seem a bit hacky.

Indeed. Seems hacky to me too.

Reply:
@highker James, how about this?

```
    /**
     * @param tableStatistics Table-level and column-level statistics. Columns are identified using ColumnHandles.
     * @param predicate Filter expression referring to columns by name
     * @param columnNames Mapping from ColumnHandles used in tableStatistics to column names used in the predicate
     * @param columnTypes Mapping from column names used in the predicate to column types
     */
    TableStatistics filterStats(
            TableStatistics tableStatistics,
            RowExpression predicate,
            ConnectorSession session,
            Map<ColumnHandle, String> columnNames,
            Map<String, Type> columnTypes);
```

Reply:
It's much better than no documentation, at the very least! I like it. ⭐️